hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Path.java

```diff
@@ -27,7 +27,7 @@
 import java.util.Optional;
 import java.util.regex.Pattern;
 
-import org.apache.avro.reflect.Stringable;
+import org.apache.hadoop.thirdparty.avro.reflect.Stringable;
```

Comment:
This could be dangerous, as we are saying that a public class can no longer be serialised through Avro.

Do you think it will be possible for us to retain the unshaded annotation as well as adding the new one? And still have everything to work without Avro on the CP?